### PR TITLE
[AIP-51] Fix calculate_env for non local executor

### PR DESCRIPTION
**Pull request by pierrejeambrun**
https://github.com/apache/airflow/pull/27941 introduced a mistake in the `calculate_env` function.

This fixes it.

**edit**: adding a few tests

### CODE REVIEW

**Review by utkarsharma2 — APPROVED** [review]

**Comment by pierrejeambrun** [comment]
cc: @o-nikolas

File: tests/cli/commands/test_standalone_command.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    LOCAL_KUBERNETES_EXECUTOR,
+    SEQUENTIAL_EXECUTOR,
+)
+
+
+class TestStandaloneCommand:
+    @pytest.mark.parametrize(
+        "conf_executor_name, conf_sql_alchemy_conn, expected_standalone_executor",
+        [
+            (LOCAL_EXECUTOR, "sqlite_conn_string", LOCAL_EXECUTOR),
```
Comment (o-nikolas) [inline tests/cli/commands/test_standalone_command.py:41]: This actually raises an interesting edge case. This being a local executor allows it to not fall into that if branch, but we'd actually want to force to the sequential executor in this case (since sqlite is the db). This isn't anything wrong with your PR, and we should likely merge it as is. But we may want to revisit this in a separate PR.

**Review by o-nikolas — APPROVED** [review]

File: tests/cli/commands/test_standalone_command.py (hunk 1 above)
Comment (pierrejeambrun) [reply tests/cli/commands/test_standalone_command.py:41]: Yes I also noticed that, you can end up with a LOCAL_EXECUTOR and an sqlite db, which doesn't sound great

Comment (pierrejeambrun) [reply tests/cli/commands/test_standalone_command.py:41]: I will open a follow up PR to rework a bit this function so this cannot happen

Comment (o-nikolas) [reply tests/cli/commands/test_standalone_command.py:41]: Technically local executor can run in a single threaded fashion too (if configured to do so with `self.parallelism = 1`) so perhaps we should leave the code as is in case someone is trying to do this.
